helper/restore_helper.go

```diff
@@ -118,12 +127,32 @@ func (r *RestoreReader) copyAllData() (int64, error) {
 	return bytesRead, err
 }
 
+func (r *RestoreReaderImpl) getReaderType() ReaderType {
+	return r.readerType
+}
+
+func (r *RestoreReaderImpl) getFileHandle() *os.File {
+	return r.fileHandle
+}
+
 type oidWithBatch struct {
 	oid   int
 	batch int
 }
 
-func doRestoreAgent() error {
+type RestoreHelper interface {
+	getRestoreDataReader(fileToRead string, objToc *toc.SegmentTOC, oidList []int) (RestoreReader, error)
+	getRestorePipeWriter(currentPipe string) (*bufio.Writer, *os.File, error)
+	checkForSkipFile(pipeFile string, tableOid int) bool
+}
+
+type RestoreHelperImpl struct{}
+
+func (RestoreHelperImpl) checkForSkipFile(pipeFile string, tableOid int) bool {
+	return utils.FileExists(fmt.Sprintf("%s_skip_%d", pipeFile, tableOid))
+}
+
+func doRestoreAgent_internal(h Helper, rh RestoreHelper) error {
```

Comment:
I think it is also better to switch to camel case to keep the code consistent
```suggestion
func doRestoreAgentInternal(h Helper, rh RestoreHelper) error {
```